ingest/expression_files/dense_ingestor.py

```diff
@@ -0,0 +1,187 @@
+"""Module for ingesting dense matrix files
+
+DESCRIPTION
+Module provides extract and transforms function for gene expression data for
+an dense matrix.
+
+"""
+import collections
+import csv
+import datetime
+import sys
+from typing import List  # noqa: F401
+
+from bson.objectid import ObjectId
+
+try:
+    from expression_files import GeneExpression
+    sys.path.append("../ingest")
+    from ingest_files import IngestFiles
+
+except ImportError:
+    # Used when importing as external package, e.g. imports in single_cell_portal code
+    from .expression_files import GeneExpression
+    sys.path.append("../ingest")
+    from ..ingest_files import IngestFiles
+
+
+class DenseIngestor(GeneExpression, IngestFiles):
+    ALLOWED_FILE_TYPES = ["text/csv",
+                          "text/plain", "text/tab-separated-values"]
+
+    def __init__(self, file_path, study_file_id, study_id, **kwargs):
+        GeneExpression.__init__(self, file_path, study_file_id, study_id)
+        IngestFiles.__init__(
+            self, file_path, allowed_file_types=self.ALLOWED_FILE_TYPES
+        )
+        self.matrix_params = kwargs
+        self.csv_file_handler, self.file_handler = self.open_file(
+            self.file_path)
+        self.gene_names = {}
+        self.header = next(self.csv_file_handler)
+
+    def execute_ingest(self):
```

Comment:
Add a method comment.  This method can only ever be called once for a given instance of DenseIngestor, right?  That might be good to add in the comment

Reply:
Isn't that implied by having self as an argument?

Reply:
No, self just lets you know that this is a class method.  It doesn't let you know that this method is very tightly coupled to the current position of the csv_file_handler, such that calling this method twice would give bad results.  And it's just generally good practice at the start of long methods to have a comment explaining what the method is intended to do